diff --git a/marimo/_server/api/endpoints/login.py b/marimo/_server/api/endpoints/login.py
index aa5e623ce91..6c9000f4cd2 100644
--- a/marimo/_server/api/endpoints/login.py
+++ b/marimo/_server/api/endpoints/login.py
@@ -128,7 +128,7 @@ async def login_submit(request: Request) -> Response:
     elif request.user.is_authenticated:
         return RedirectResponse(redirect_url, 302)
 
-    base_url = AppState(request).base_url
+    base_url = _with_trailing_slash(AppState(request).base_url or "/")
     html = LOGIN_PAGE.format(error=error, base_url=base_url)
     return HTMLResponse(
         content=html,
