diff --git a/pytype/pyi/parser_test.py b/pytype/pyi/parser_test.py
index 36a7a6e73..6499705e1 100644
--- a/pytype/pyi/parser_test.py
+++ b/pytype/pyi/parser_test.py
@@ -576,9 +576,9 @@ class A: ...
 
   def test_def(self):
     self.check("""
-      def f(x: "int") -> "str": ...
+      def f(x: "int", *args: "float", y: "str", **kwargs: "bool") -> "str": ...
     """, """
-      def f(x: int) -> str: ...
+      def f(x: int, *args: float, y: str, **kwargs: bool) -> str: ...
     """)
 
 
@@ -846,6 +846,13 @@ def test_params(self):
     self.check("def foo(x: int, y: str, z: bool,) -> int: ...",
                "def foo(x: int, y: str, z: bool) -> int: ...")
 
+  def test_defaults(self):
+    self.check("""
+      def f(x: int = 3, y: str = " ") -> None: ...
+    """, """
+      def f(x: int = ..., y: str = ...) -> None: ...
+    """)
+
   def test_star_params(self):
     self.check("def foo(*, x) -> str: ...")
     self.check("def foo(x: int, *args) -> str: ...")
